visualmetrics/visualmetrics-portable.py

```diff
@@ -1346,6 +1371,90 @@ def load_histograms(histograms_file, start, end):
     return histograms
 
 
+def calculate_key_color_frames(histograms, key_colors):
+    if not key_colors:
+        return {}
+
+    key_color_frames = {}
+    for key in key_colors:
+        key_color_frames[key] = []
+
+    current = None
+    current_key = None
+    total = 0
+    matched = 0
+    channels = ["r", "g", "b"]
+    for index, histogram in enumerate(histograms):
```

Comment:
Is there any specific reason you chose to use a `for` loop here? The matching/current_key seems a bit over-complicated to me, and it could be simplified by using a couple while loops instead, here's some pseudo-code of what I'm thinking about:
```
        while histograms left:
            check if current histogram matches a key
            if not, continue to next histogram

            while we haven't found the final histogram for this key:
                check if current histogram matches
                    add to framecount
                if not, break out

            set the end time of the key
```